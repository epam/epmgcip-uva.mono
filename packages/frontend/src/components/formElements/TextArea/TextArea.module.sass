@use 'src/style/colors'
@import 'src/style/typography'

.label
    @extend .H6
    display: flex
    flex-direction: column
    margin-bottom: 18px

.inputRequired
    color: colors.$red

.textArea
    @extend .T1
    font-family: Inter
    width: calc(100vw - 48px)
    max-width: calc(768px - 48px)
    height: 148px
    border: 1px solid colors.$gray-200
    border-radius: 8px
    background-color: colors.$white
    padding: 12px
    box-sizing: border-box
    margin-top: 8px
    resize: none

    &::placeholder
        color: colors.$gray-200

.inputError
    border-color: colors.$red

.validationError
    @extend .T3
    align-self: flex-start
    color: colors.$red
    margin-top: 2px
    margin-bottom: 0