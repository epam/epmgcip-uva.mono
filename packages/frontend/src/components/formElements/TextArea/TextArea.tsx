import { getClassesList } from 'src/utils/getClassesList';
import css from './TextArea.module.sass';
import translation from 'src/translations/Russian.json';

interface TextAreaProps {
  value: string;
  setChange: React.Dispatch<React.SetStateAction<string>>;
  labelText?: string;
  required?: boolean;
  placeholder?: string;
  labelClassName?: string;
  textAreaClassName?: string;
  isValidationError?: boolean;
  errorMessage?: string;
}

export const TextArea = ({
  value,
  setChange,
  labelText,
  required = false,
  placeholder,
  labelClassName,
  textAreaClassName,
  isValidationError,
  errorMessage,
}: TextAreaProps) => {
  const labelClasses = getClassesList(css.label, labelClassName);

  const textAreaClasses = getClassesList(
    css.textArea,
    textAreaClassName,
    isValidationError ? css.inputError : undefined
  );

  const handleSetChange = (e: React.ChangeEvent<HTMLTextAreaElement>) => {
    setChange(e.target.value);
  };

  return (
    <label className={labelClasses}>
      {labelText && (
        <div>
          {labelText}
          {required && <span className={css.inputRequired}> *</span>}
        </div>
      )}
      <textarea
        placeholder={placeholder || translation.input}
        className={textAreaClasses}
        value={value}
        onChange={(e) => handleSetChange(e)}
      />
      {isValidationError && (
        <p className={css.validationError}>
          {isValidationError && errorMessage}
        </p>
      )}
    </label>
  );
};
