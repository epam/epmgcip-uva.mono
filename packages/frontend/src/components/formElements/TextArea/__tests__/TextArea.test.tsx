import { describe, expect, it, vi } from 'vitest';
import { fireEvent, render, screen } from '@testing-library/react';
import { TextArea } from '../TextArea';

const valueMock = 'Test Value';
const labelTextMock = 'Test Label';
const setChangekMock = vi.fn();

describe('Testing: TextArea', () => {
  it('should render', () => {
    const { container } = render(
      <TextArea
        value={valueMock}
        setChange={setChangekMock}
        labelText={labelTextMock}
      />
    );

    expect(container).toMatchSnapshot();
  });

  it('should call provided function after input text', () => {
    render(
      <TextArea
        value={valueMock}
        setChange={setChangekMock}
        labelText={labelTextMock}
      />
    );

    const renderedButton = screen.getByDisplayValue(valueMock);
    const expected = 'Expected Text';

    fireEvent.change(renderedButton, { target: { value: expected } });

    expect(setChangekMock).toHaveBeenCalledWith(expected);
  });
});
