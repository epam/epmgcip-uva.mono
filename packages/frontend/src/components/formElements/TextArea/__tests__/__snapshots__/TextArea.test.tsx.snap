// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Testing: TextArea > should render 1`] = `
<div>
  <label
    class="_label_6edfa1"
  >
    <div>
      Test Label
    </div>
    <textarea
      class="_textArea_6edfa1"
      placeholder="Ввод"
    >
      Test Value
    </textarea>
  </label>
</div>
`;
